mobilecoind/src/service.rs

```diff
@@ -1418,12 +1418,21 @@ impl<
             })
             .collect::<Result<Vec<Outlay>, RpcStatus>>()?;
 
+        // Set change address to sender address unless it has been overridden
+        let change_subaddress = {
```

Comment:
I don't think the extra block is needed. this can be `let change_subaddress = if request ... {`